simpleline/event_loop/main_loop.py

```diff
@@ -181,13 +181,21 @@ def process_signals(self, return_after=None):
         :param return_after: Wait on this signal to be processed.
         :type return_after: Class of the signal.
         """
+        if return_after is not None:
+            self._process_signals_with_return(return_after)
+        else:
+            self._process_signals_iteration()
+
+    def _process_signals_with_return(self, return_after):
+        """Process signals until the return_after signal was processed.
+
+        Or the loop quited.
+        """
         # get unique ID when waiting for the signal
         unique_id = self._register_wait_on_signal(return_after)
 
-        while self._can_process_signals() or return_after is not None:
+        while self._run_loop or return_after is not None:
```

Comment:
We know that `return_after` is not None here, no?

Reply:
True, fixed.